pandas/io/formats/style.py

```diff
@@ -987,34 +987,95 @@ def set_caption(self, caption: str) -> "Styler":
         self.caption = caption
         return self
 
-    def set_table_styles(self, table_styles) -> "Styler":
+    def set_table_styles(self, table_styles, axis=0, overwrite=True) -> "Styler":
```

Comment:
can you type table_styles

Reply:
I tried adding the following to func def, but it got errors, I don't know my way around mypy and this seems slightly more complicated than normal with the multiple acceptable input types and programming logic:

    table_styles: Union[
            List[Dict[str, Union[str, List[Tuple[str, str]]]]],
            Dict[
                Union[str, Tuple[Any, ...]],
                List[Dict[str, Union[str, List[Tuple[str, str]]]]],
            ],
        ] = []

So I reverted the attempt.

However, all green, pinging @jreback